modules/apps/29-fee/keeper/relay.go

```diff
@@ -0,0 +1,31 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/cosmos/ibc-go/v3/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+)
+
+// SendPacket wraps IBC ChannelKeeper's SendPacket function
+func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	return k.ics4Wrapper.SendPacket(ctx, chanCap, packet)
+}
+
+// WriteAcknowledgement wraps IBC ChannelKeeper's WriteAcknowledgement function
+// ICS29 WriteAcknowledgement is used for asynchronous acknowledgements
+func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement []byte) error {
+	// retrieve the forward relayer that was stored in `onRecvPacket`
+	packetId := channeltypes.NewPacketId(packet.GetSourceChannel(), packet.GetSourcePort(), packet.GetSequence())
+	relayer, _ := k.GetForwardRelayerAddress(ctx, packetId)
+
+	k.DeleteForwardRelayerAddress(ctx, packetId)
+
+	ack := types.NewIncentivizedAcknowledgement(relayer, acknowledgement)
```

Comment:
If my understanding on the [core logic](https://github.com/cosmos/ibc-go/blob/ics29-fee-middleware/modules/core/keeper/msg_server.go#L516-L531) is correct, the `acknowledgement` we get here is already serialized from the `IncentivizedAcknowledgement` returned from `OnRecvPacket` by the fee middleware. So this would result in double wrapping of the `acknowledgement` bytes, and the inner module at the counterparty chain would not able to parse the inner acknowledgement correctly.